go/kbfs/libfs/node_wrappers.go

```diff
@@ -54,20 +89,76 @@ type specialFileNode struct {
 var _ libkbfs.Node = (*specialFileNode)(nil)
 
 var perTlfWrappedNodeNames = map[string]bool{
-	StatusFileName: true,
+	StatusFileName:        true,
+	UpdateHistoryFileName: true,
+}
+
+var perTlfWrappedNodePrefixes = []string{
+	UpdateHistoryFileName,
+}
+
+func shouldBeTlfWrappedNode(name string) bool {
+	for _, p := range perTlfWrappedNodePrefixes {
+		if strings.HasPrefix(name, p) {
+			return true
+		}
+	}
+	return perTlfWrappedNodeNames[name]
+}
+
+func (sfn *specialFileNode) newUpdateHistoryFileNode(
+	node libkbfs.Node, name string) *updateHistoryFileNode {
+	revs := strings.TrimPrefix(name, UpdateHistoryFileName)
+	if revs == "" {
+		return &updateHistoryFileNode{
+			Node:   node,
+			fb:     sfn.GetFolderBranch(),
+			config: sfn.config,
+			log:    sfn.log,
+			start:  kbfsmd.RevisionInitial,
+			end:    kbfsmd.RevisionUninitialized,
+		}
+	}
+
+	matches := updateHistoryRevsRE.FindStringSubmatch(revs)
+	if len(matches) != 4 {
+		return nil
+	}
+
+	start, err := strconv.ParseUint(matches[1], 10, 64)
+	if err != nil {
+		return nil
+	}
+	end := start
+	if matches[3] != "" {
+		end, err = strconv.ParseUint(matches[3], 10, 64)
+		if err != nil {
+			return nil
+		}
+	}
+
+	return &updateHistoryFileNode{
+		Node:   node,
+		fb:     sfn.GetFolderBranch(),
+		config: sfn.config,
+		log:    sfn.log,
+		start:  kbfsmd.Revision(start),
+		end:    kbfsmd.Revision(end),
+	}
 }
 
 // ShouldCreateMissedLookup implements the Node interface for
 // specialFileNode.
 func (sfn *specialFileNode) ShouldCreateMissedLookup(
 	ctx context.Context, name data.PathPartString) (
 	bool, context.Context, data.EntryType, os.FileInfo, data.PathPartString) {
-	if !perTlfWrappedNodeNames[name.Plaintext()] {
+	plain := name.Plaintext()
+	if !shouldBeTlfWrappedNode(plain) {
 		return sfn.Node.ShouldCreateMissedLookup(ctx, name)
 	}
 
-	switch name.Plaintext() {
-	case StatusFileName:
+	switch {
+	case plain == StatusFileName:
```

Comment:
I was trying to think of a way to integrate the `perTlfWrappedNodeNames` and `perTlfWrappedNodePrefixes` here, perhaps by having them be a map to function handlers, but actually I don't think it would make the code any cleaner.